Firestore/core/src/firebase/firestore/api/firestore.mm

```diff
@@ -61,31 +61,35 @@
       persistence_key_{std::move(persistence_key)},
       worker_queue_{std::move(worker_queue)},
       extension_{extension} {
-  settings_ = [[FIRFirestoreSettings alloc] init];
 }
 
 AsyncQueue* Firestore::worker_queue() {
   return [client_ workerQueue];
 }
 
-FIRFirestoreSettings* Firestore::settings() const {
+const Settings& Firestore::settings() const {
   std::lock_guard<std::mutex> lock{mutex_};
-  // Disallow mutation of our internal settings
-  return [settings_ copy];
+  return settings_;
 }
 
-void Firestore::set_settings(FIRFirestoreSettings* settings) {
+void Firestore::set_settings(Settings&& settings) {
   std::lock_guard<std::mutex> lock{mutex_};
-  // As a special exception, don't throw if the same settings are passed
-  // repeatedly. This should make it more friendly to create a Firestore
-  // instance.
-  if (client_ && ![settings_ isEqual:settings]) {
+  if (client_) {
     HARD_FAIL(
         "Firestore instance has already been started and its settings can "
         "no longer be changed. You can only set settings before calling any "
         "other methods on a Firestore instance.");
   }
-  settings_ = [settings copy];
+  settings_ = std::move(settings);
+}
+
+void Firestore::set_user_executor(
+    std::unique_ptr<util::Executor>&& user_executor) {
```

Comment:
`std::unique_ptr` is already a move-only type so you don't need to mark this as an rvalue-reference.